app/models/task.py

```diff
@@ -3,4 +3,29 @@
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    __tablename__= "tasks"
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title=db.Column(db.String(200))
+    description=db.Column(db.String(200))
+    completed_at=db.Column(db.DateTime, nullable=True)
+    goal_id=db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    
+    def to_dict(self):
+        if self.goal_id:
+            return {
+            "id":self.task_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": True if self.completed_at else False,
+            "goal_id": self.goal_id
+        }
+
+        else:
+            return{
+            "id":self.task_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": True if self.completed_at else False
+        }
```

Comment:
This definitely works! Can you think of a way to restructure it so we set up most of the dictionary outside of the if/else and only add the `"goal_id"` key when needed? This could help reduce repetition in your code.